src/pipecat/services/simli.py

```diff
@@ -0,0 +1,116 @@
+        self.ready = True
+        # Create task to consume and process audio and video
+        self.AudioTask = asyncio.create_task(self.consume_and_process_audio())
+        self.VideoTask = asyncio.create_task(self.consume_and_process_video())
+
+    async def consume_and_process_audio(self):
+        async for audio_frame in self.simliClient.getAudioStreamIterator():
+            # Process the audio frame
+            try:
+                resampledFrames = self.pipecatResampler.resample(audio_frame)
+                for resampled_frame in resampledFrames:
+                    await self.push_frame(
+                        TTSAudioRawFrame(
+                            audio=resampled_frame.to_ndarray().tobytes(),
+                            sample_rate=self.pipecatResampler.rate,
+                            num_channels=1,
+                        ),
+                    )
+            except Exception as e:
+                print(e)
+                import traceback
+
+                traceback.print_exc()
+
+    async def consume_and_process_video(self):
+        async for video_frame in self.simliClient.getVideoStreamIterator(
+            targetFormat="rgb24"
+        ):
+            # Process the video frame
+            convertedFrame: OutputImageRawFrame = OutputImageRawFrame(
+                image=video_frame.to_rgb().to_image().tobytes(),
+                size=(video_frame.width, video_frame.height),
+                format="RGB",
+            )
+            convertedFrame.pts = video_frame.pts
+            await self.push_frame(
+                convertedFrame,
+            )
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        await super().process_frame(frame, direction)
+
+        if isinstance(frame, TTSAudioRawFrame):
+            # Send audio frame to Simli
+            try:
+                if self.ready:
+                    AudioFrame
+                    oldFrame = AudioFrame.from_ndarray(
+                        np.frombuffer(frame.audio, dtype=np.int16)[None, :],
+                        layout=frame.num_channels,
+                    )
+                    oldFrame.sample_rate = frame.sample_rate
+                    if self.pipecatResampler is None:
+                        self.pipecatResampler = AudioResampler(
+                            "s16", oldFrame.layout, oldFrame.sample_rate
+                        )
+
+                    resampledFrame = self.simliResampler.resample(oldFrame)
+                    for frame in resampledFrame:
+                        await self.simliClient.send(
+                            frame.to_ndarray().astype(np.int16).tobytes()
+                        )
+                    return
+                else:
+                    print(
+                        "Simli Connection is not Initialized properly, passing audio to next processor"
+                    )
+                    await self.push_frame(frame, direction)
```

Comment:
I think we can just do:

```
await self.push_frame(frame, direction)
```

That is, no need to check or print anything. This way we pass the audio to the downstream processors in case they want to use it.

Reply:
Oh, actually you already have that at the end of the function, so you just don't need the `else` block.